redshift_sqlalchemy/dialect.py

```diff
@@ -700,70 +700,189 @@ def _get_all_constraint_info(self, connection, **kw):
         return self._all_constraints
 
 
+# At the time of this implementation, no specification for a session token was
+# found. After looking at a few session tokens they appear to be the same as
+# the aws_secret_access_key pattern, but much longer. An example token can be
+# found here:
+#   http://docs.aws.amazon.com/STS/latest/APIReference/API_GetSessionToken.html
+# The regexs for access keys can be found here:
+#     http://blogs.aws.amazon.com/security/blog/tag/key+rotation
+creds_rx = re.compile(r"""
+    ^aws_access_key_id=[A-Z0-9]{20};
+    aws_secret_access_key=[A-Za-z0-9/+=]{40}
+    (?:;token=[A-Za-z0-9/+=]+)?$
+""", re.VERBOSE)
+
+
+def process_aws_credentials(access_key_id, secret_access_key,
+                            session_token=None):
+    credentials = 'aws_access_key_id={0};aws_secret_access_key={1}'.format(
+        access_key_id,
+        secret_access_key
+    )
+
+    if session_token is not None:
+        credentials += ';token={0}'.format(session_token)
+
+    if not creds_rx.match(credentials):
+        raise ValueError(
+            'credentials must match the following'
+            ' format:\n'
+            'aws_access_key_id=<access-key-id>;'
+            'aws_secret_access_key=<secret-access-key>'
+            '[;token=<temporary-session-token>]\ngot %r' % credentials
+        )
```

Comment:
I don't really understand why you're formatting something here and then immediately checking whether it's formatted correctly? Can't you just make sure to format it right?